### PR TITLE
feat(gh-prov-refactor-cdx-commit): removed external mongo service in Github Service and refactored GitHub Provider to use commit hashes as database key for Commits instead of the URL of the repository

### PR DESCRIPTION
**Pull request by qtpeters**
## Summary

Refactored GitHub Provider and service to remove the external Mongo Service and made the Commit's DB key the last hash rather than the URL.  This caused a fairly significant refactor.

### Changed

- The Commit DB key to the last hash in the repo

### Removed

- The external Mongo Service in the GitHub Provider. Moved functionality into the GitHub service proper.

## How to test

1. Set up test environment
2. Run `cargo test`
3. Run `./target/debug/harbor ingest -p github harbor-test-org --debug`
  - Verify that all of the repos are processed as usual
  - Run the command a second time and verify that all of the commit hashes extracted from the previous run are 
    found and no repos are processed redundantly.
  - Extras credit: Change a file in one of the repos under the `harbor-test-org`, commit and push the change. 
    Run the harbor command again and verify that _only_ that repo is processed. 
4. Open Compass and look at the data:  
    - In the `Commit` collection, the `id` will now be a string identifiable as a Git Commit hash rather than the URL.
    - In the `Package` collection, the results will be the same as past runs

### CODE REVIEW

File: sdk/core/src/services/github/service.rs [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+use platform::persistence::mongodb::Service as MongoService;
+use platform::Error as PlatformError;
+
 
 /// Definition of the GitHubProvider
 #[derive(Debug)]
 pub struct GitHubService {
     org: String,
     client: GitHubClient,
+    pub(crate) store: Arc<Store>
```
Comment (DerekStrickland) [inline sdk/core/src/services/github/service.rs:18]: Pub for tests I assume?

File: sdk/core/src/services/github/service.rs [hunk 2]
```diff
@@ -114,11 +117,33 @@ impl GitHubService {
...
 }
 
+#[async_trait]
+impl MongoService<Commit> for GitHubService {
+    fn store(&self) -> Arc<Store> {
+        self.store.clone()
+    }
+
+    /// Insert a document into a [Collection].
+    async fn insert<'a>(&self, doc: &mut Commit) -> Result<(), PlatformError> {
```
Comment (DerekStrickland) [inline sdk/core/src/services/github/service.rs:127]: Worth doing some validation here, since the ID gets set elsewhere?

**Review by DerekStrickland — COMMENTED** [review]
Couple comments.  Almost there I think.

File: sdk/core/src/services/github/service.rs (hunk 1 above)
Comment (qtpeters) [reply sdk/core/src/services/github/service.rs:18]: Yeah... Maybe but let me see.

File: sdk/core/src/services/github/service.rs (hunk 2 above)
Comment (qtpeters) [reply sdk/core/src/services/github/service.rs:127]: You bet, I'll get in there

File: sdk/core/src/services/github/service.rs (hunk 1 above)
Comment (qtpeters) [reply sdk/core/src/services/github/service.rs]: Nope!  Turns out I didn't need to keep it public.  Great catch

**Review by DerekStrickland — APPROVED** [review]
LGTM

**Pull request merged**